webui/react/src/themes.ts

```diff
@@ -154,26 +158,38 @@ export const lightTheme: Theme = {
 };
 
 const stateColorMapping = {
-  [RunState.Active]: 'var(--theme-colors-states-active)',
-  [RunState.Canceled]: 'var(--theme-colors-states-inactive)',
-  [RunState.Completed]: 'var(--theme-colors-states-success)',
-  [RunState.Deleted]: 'var(--theme-colors-states-failed)',
-  [RunState.Errored]: 'var(--theme-colors-states-failed)',
-  [RunState.Paused]: 'var(--theme-colors-states-suspended)',
-  [RunState.StoppingCanceled]: 'var(--theme-colors-states-inactive)',
-  [RunState.StoppingCompleted]: 'var(--theme-colors-states-success)',
-  [RunState.StoppingError]: 'var(--theme-colors-states-failed)',
-  [CommandState.Pending]: 'var(--theme-colors-states-suspended)',
-  [CommandState.Assigned]: 'var(--theme-colors-states-suspended)',
-  [CommandState.Pulling]: 'var(--theme-colors-states-active)',
-  [CommandState.Starting]: 'var(--theme-colors-states-active)',
-  [CommandState.Running]: 'var(--theme-colors-states-active)',
-  [CommandState.Terminating]: 'var(--theme-colors-states-inactive)',
-  [CommandState.Terminated]: 'var(--theme-colors-states-inactive)',
+  [RunState.Active]: 'active',
+  [RunState.Canceled]: 'inactive',
+  [RunState.Completed]: 'success',
+  [RunState.Deleted]: 'failed',
+  [RunState.Errored]: 'failed',
+  [RunState.Paused]: 'suspended',
+  [RunState.StoppingCanceled]: 'inactive',
+  [RunState.StoppingCompleted]: 'success',
+  [RunState.StoppingError]: 'failed',
+  [CommandState.Pending]: 'suspended',
+  [CommandState.Assigned]: 'suspended',
+  [CommandState.Pulling]: 'transitioning',
+  [CommandState.Starting]: 'transitioning',
+  [CommandState.Running]: 'active',
+  [CommandState.Terminating]: 'inactive',
+  [CommandState.Terminated]: 'inactive',
+  [ResourceState.Free]: 'inactive',
+};
+
+export const getStateColorCssVar = (
+  state: RunState | CommandState | ResourceState | undefined,
+): string => {
+  const name = state ? stateColorMapping[state] : 'active';
+  return `var(--theme-colors-states-${name})`;
 };
 
-export const getStateColor = (state: RunState | CommandState | undefined): string => {
-  return stateColorMapping[state || RunState.Active];
+export const getStateColor = (
+  state: RunState | CommandState | ResourceState,
+  theme = lightTheme,
+): string => {
+  const name = state ? stateColorMapping[state] : 'active';
+  return theme.colors.states[name as StateColors];
```

Comment:
**suggestion:** based on [this post](https://stackoverflow.com/questions/41725725/access-css-variable-from-javascript) we can convert CSS var to the actual value:
```
export const getStateColor = (
  state: RunState | CommandState | ResourceState | undefined,
): string => {
  const name = state ? stateColorMapping[state] : 'active';
  return getComputedStyle(document.body).getPropertyValue(`--theme-colors-states-${name}`)
};
```

Reply:
neat! https://caniuse.com/#search=getComputedStyle

Reply:
**question:** did you want to add this to this PR?

Reply:
yeah for sure. it wasn't just assigned to me yet so was waiting for the review to finish : )